### PR TITLE
[Nhk] Fix: NhkVod(|Program) Extractor  / [Piksel] Feat: Add variable host support

### PR DESCRIPTION
**Pull request by menschel**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

NHK World VOD is broken since a few weeks.

This is due changing to self-hosted API and player instance.

Necessary tweaks were to
- make PikselIE use the host of the given URL instead of hard-coded Piksel host.
- use the correct template url for the initial API call in NhkVodIE

### CODE REVIEW

**Comment by menschel** [comment]
Honestly, I hope this does not result in nitpicking again!
Take it or drop it.

**Comment by dirkf** [comment]
>... again ??

File: youtube_dl/extractor/nhk.py [hunk 1]
```diff
@@ -21,6 +21,78 @@ def _call_api(self, m_id, lang, is_video, is_episode, is_clip):
...
+            'id': 'yd8322ch',
+            'ext': 'mp4',
+            'description': 'NHK WORLD-JAPAN presents a sumo highlights program for fans around the globe. Today the'
+                           ' show features all top-division bouts from May 14, Day 1 of the Grand Sumo Tournament in'
+                           ' Tokyo.',
+            'title': 'GRAND SUMO Highlights - [Recap] May Tournament Day 1 (Opening Day)',
+            'upload_date': '20230514',
+            'timestamp': 1684083791,
+        },
+
```
Comment (dirkf) [inline youtube_dl/extractor/nhk.py:42]:
```suggestion
```

File: youtube_dl/extractor/piksel.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
     parse_iso8601,
     try_get,
     unescapeHTML,
+    urljoin
```
Comment (dirkf) [inline youtube_dl/extractor/piksel.py:15]:
```suggestion
    urljoin,
```

File: youtube_dl/extractor/piksel.py [hunk 3]
```diff
@@ -73,10 +74,9 @@ def _extract_url(webpage):
         if mobj:
             return mobj.group('url')
 
-    def _call_api(self, app_token, resource, display_id, query, fatal=True):
-        response = (self._download_json(
-            'http://player.piksel.com/ws/ws_%s/api/%s/mode/json/apiv/5' % (resource, app_token),
-            display_id, query=query, fatal=fatal) or {}).get('response')
+    def _call_api(self, app_token, resource, display_id, query, host="https://player.piksel.com", fatal=True):
```
Comment (dirkf) [inline youtube_dl/extractor/piksel.py:77]:
```suggestion
    def _call_api(self, app_token, resource, display_id, query, host='https://player.piksel.com', fatal=True):
```

File: youtube_dl/extractor/piksel.py [hunk 4]
```diff
@@ -73,10 +74,9 @@ def _extract_url(webpage):
...
+        url = urljoin(base=host, path='/ws/ws_%s/api/%s/mode/json/apiv/5' % (resource, app_token))
+        response = (self._download_json(url, display_id, query=query, fatal=fatal) or {}).get('response')
```
Comment (dirkf) [inline youtube_dl/extractor/piksel.py:79]: We could ensure that a `dict` is always returned, even if the returned JSON isn't one (add `traverse_obj` to utils imports). Also, no need to use kwargs for `urljoin()`:
```suggestion
        url = urljoin(host, '/ws/ws_%s/api/%s/mode/json/apiv/5' % (resource, app_token))
        response = self._download_json(url, display_id, query=query, fatal=fatal)
        response = traverse_obj(response, 'response', expected_type=dict) or {}
```

**Review by dirkf — CHANGES_REQUESTED** [review]
Thanks for your work!

Looks good: I've made some small suggestions for project conventions and robustness.

**Review by menschel — COMMENTED** [review]
Accepted changes.

File: youtube_dl/extractor/piksel.py [hunk 5]
```diff
@@ -12,7 +12,7 @@
     parse_iso8601,
     try_get,
```
Comment (dirkf) [inline youtube_dl/extractor/piksel.py:13]:
```suggestion
    traverse_obj,
    try_get,
```

File: youtube_dl/extractor/piksel.py [hunk 6]
```diff
@@ -11,6 +11,7 @@
     int_or_none,
     parse_iso8601,
     traverse_obj,
+    traverse_obj,
```
Comment (menschel) [inline youtube_dl/extractor/piksel.py:14]: This is now duplicate ?!

**Comment by menschel** [comment]
To say it bluntly, this code review process did not yield any significant quality improvement.
It merely wasted 1 week and introduced a flake8 issue along the way.
Solving this simple duplicate import made me realize how Github has slided down in Quality after Micro$oft has taken over Regime. Please move to another platform.

**Pull request closed without merging**